### PR TITLE
Add missing methods to HTTPX sync clients

### PR DESCRIPTION
**Pull request by ulodciv**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Other, please describe:

---

- [x] You consent that the copyright of your pull request source code belongs to Authlib's author.

### CODE REVIEW

**Comment by lepture** [comment]
We don't need them, they are already in the parent class.

**Comment by ulodciv** [comment]
@lepture I hear you, but these methods are present in the async clients, and I see no obvious reason why they would be absent from the sync clients. Furthermore, they don't work in my code as is, whereas they work with my version of them, which is exactly the same, except for the additional methods I have. 

I had them included in the the PR https://github.com/lepture/authlib/pull/192 I've been using them, and still have to use them since the ones that were added recently don't work.

File: authlib/integrations/httpx_client/assertion_client.py [hunk 1]
```diff
@@ -87,3 +87,14 @@ def request(self, method, url, withhold_token=False, auth=None, **kwargs):
             auth = self.token_auth
         return super(AssertionClient, self).request(
             method, url, auth=auth, **kwargs)
+    
+    def _refresh_token(self, data):
```
Comment (lepture) [inline authlib/integrations/httpx_client/assertion_client.py:91]: see: https://github.com/lepture/authlib/blob/master/authlib/oauth2/rfc7521/client.py#L72

File: authlib/integrations/httpx_client/oauth2_client.py [hunk 2]
```diff
@@ -212,3 +212,47 @@ def ensure_active_token(self):
                 self.update_token(token, access_token=access_token)
         else:
             raise InvalidTokenError()
+
+    def _fetch_token(
```
Comment (lepture) [inline authlib/integrations/httpx_client/oauth2_client.py:216]: see: https://github.com/lepture/authlib/blob/master/authlib/oauth2/client.py#L207

**Comment by lepture** [comment]
> they don't work in my code as is

Is it possible to create a test case in the `tests` folder?

**Pull request closed without merging**